modules/codegen/src/main/scala/com/twilio/guardrail/generators/CirceProtocolGenerator.scala

```diff
@@ -381,7 +381,7 @@ object CirceProtocolGenerator {
       case ExtractSuperClass(swagger, definitions) =>
         def allParents: Tracker[Schema[_]] => Target[List[(String, Tracker[Schema[_]], List[Tracker[Schema[_]]])]] =
           _.refine[Target[List[(String, Tracker[Schema[_]], List[Tracker[Schema[_]]])]]]({ case x: ComposedSchema => x })(
-            _.downField("allOf", _.getAllOf()).indexedDistribute match {
+            _.downField("allOf", _.getAllOf().asScala.toList.filter(s => s.get$ref() != null)).indexedDistribute match {
```

Comment:
```suggestion
            _.downField("allOf", _.getAllOf()).indexedDistribute.filter(_.downField("$ref", _.get$ref()).unwrapTracker.nonEmpty) match {
```

Reply:
Pretty sure this will work, I'm on mobile right now so I can't test. Leaving the comment in case you get to it before I do.

Reply:
What's nice about it is that `indexedDistribute` happens before the filter, so array index is preserved instead of being the index into items after filtering

Reply:
(this is part of the new infrastructure introduced in #402, so I'm not expecting everyone to know how it works yet)

Reply:
thanks for the comment, i will apply it later today